platform-modules/content-manager/src/main/java/org/ekstep/content/util/PublishFinalizeUtil.java

```diff
@@ -25,29 +26,55 @@ public String uploadFile(String fileUrl, Node node, String basePath) {
 			String newFileName = node.getIdentifier() + "_" + System.currentTimeMillis() + "." + FilenameUtils.getExtension(file.getName());
     		boolean renameStatus = file.renameTo(new File(file.getParentFile(),newFileName));
 			
-		try {
-			String folder = S3PropertyReader.getProperty(CONTENT_FOLDER);
-			folder = folder + "/" + Slug.makeSlug(node.getIdentifier(), true) + "/" + S3PropertyReader.getProperty(ARTEFACT_FOLDER);
-			String[] url = null;
-			if(renameStatus)
-				url = CloudStore.uploadFile(folder, new File(file.getParent() + "/" + newFileName), true);
-			else
-				url = CloudStore.uploadFile(folder, file, true);
-			return url[1];
-		} catch (Throwable e) {
-			TelemetryManager.error("Error during uploading question paper pdf file for content:: " + node.getIdentifier() + " Error:: " + e.getStackTrace());
-			throw new ServerException(TaxonomyErrorCodes.SYSTEM_ERROR.name(),
-                    "Error during uploading question paper pdf file for content :" + node.getIdentifier()+". Please Try Again After Sometime!");
-		} finally {
 			try {
-				if(null != file)
-					delete(file);
-				
-			} catch (IOException e) {
-				e.printStackTrace();
+				String folder = S3PropertyReader.getProperty(CONTENT_FOLDER);
+				folder = folder + "/" + Slug.makeSlug(node.getIdentifier(), true) + "/" + S3PropertyReader.getProperty(ARTEFACT_FOLDER);
+				String[] url = null;
+				if(renameStatus)
+					url = CloudStore.uploadFile(folder, new File(file.getParent() + "/" + newFileName), true);
+				else
+					url = CloudStore.uploadFile(folder, file, true);
+				return url[1];
+			} catch (Throwable e) {
+				TelemetryManager.error("Error during uploading question paper pdf file for content:: " + node.getIdentifier() + " Error:: " + e.getStackTrace());
+				throw new ServerException(TaxonomyErrorCodes.SYSTEM_ERROR.name(),
+	                    "Error during uploading question paper pdf file for content :" + node.getIdentifier()+". Please Try Again After Sometime!");
+			} finally {
+				try {
+					if(null != file)
+						delete(file);
+					
+				} catch (IOException e) {
+					e.printStackTrace();
+				}
+			}
+		}
+		return null;
+	}
+	
+	public void replaceArtifactUrl(Node node) {
+		String artifactBasePath = (String)node.getMetadata().get("artifactBasePath");
```

Comment:
since this method is same for both publish pipeline and asset enrichment, can we move this to BaseFinalizer